### PR TITLE
move location checking to validate step

### PR DESCRIPTION
**Pull request by paulmey**
Add a check for location and getting rid of static list of instance sizes

@ahmetalpbalkan please review

### CODE REVIEW

File: packer/builder/azure/step_validate.go [hunk 1]
```diff
@@ -26,9 +27,42 @@ func (*StepValidate) Run(state multistep.StateBag) multistep.StepAction {
...
 	ui.Say("Validating Azure options...")
 
+	locationsResponse, err := location.NewClient(client).ListLocations()
+	if err != nil {
+		err = fmt.Errorf("Error checking location: %v", err)
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+	if err := func() error {
```
Comment (ahmetalpbalkan) [inline packer/builder/azure/step_validate.go]: -1 for this syntax, instead create an inline function

**Comment by boumenot** [comment]
This is a very old PR, what do we need to do to address it?

* Close it?
* Fix, and re-issue the PR?

**Comment by paulmey** [comment]
I’ll fix and update this PR

**Comment by paulmey** [comment]
Will re-open

**Pull request closed without merging**